hardware_interface/include/hardware_interface/actuator_interface.hpp

```diff
@@ -231,6 +405,26 @@ class ActuatorInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNod
     lifecycle_state_ = new_state;
   }
 
+  void set_state(const std::string & interface_name, const double & value)
+  {
+    actuator_states_.at(interface_name)->set_value(value);
+  }
+
+  double get_state(const std::string & interface_name) const
+  {
+    return actuator_states_.at(interface_name)->get_value();
+  }
+
+  void set_command(const std::string & interface_name, const double & value)
+  {
+    actuator_commands_.at(interface_name)->set_value(value);
+  }
+
+  double get_command(const std::string & interface_name) const
+  {
+    return actuator_commands_.at(interface_name)->get_value();
+  }
+
```

Comment:
When we use the backward compatibility part of code, we can't use these methods (which is completely valid)

I don't know if it wiser to expose if they don't work for the earlier approach

So, we can do 2 things here:

* Remove support for old way of exposing interfaces completely, so we can make use of these methods
* Remove these methods along with some changes with above comments I mentioned, in that way the user will have the liberty to choose the way he accesses the interfaces via unordered_map, map or vector

It's also because all the above new methods are pretty repetitive to all Component Interfaces (Actuator, Sensor and System)

Reply:
Ah thats a valid point. How would you flag up warnings?

Reply:
May be we can check if they are empty, and simply print a log message and return?
I can't think of a better idea